domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+
+        # if updated delta_az is negative, direction is CCW
+        if delta_az < 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_ccw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count >= target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                else:
+                    # micro break to spare the little rpi cpu
+                    time.sleep(0.1)
+            self._stop_moving()
+            # compare original count to current, just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+
+    def calibrate_dome_encoder_counts(self, num_cal_rotations=2):
+        """
+        Calibrate the encoder (determine degrees per tick).
+
+        Parameters
+        ----------
+        num_cal_rotations : integer
+            Number of rotations to perform to calibrate encoder.
+
+        """
+        # rotate the dome until we hit home, to give reference point
+        self.find_home()
+
+        # now set dome to rotate n times so we can determine the number of
+        # ticks per revolution
+        rotation_count = 0
+        # pause to let things settle/get a noticeable blink of debug_lights
+        time.sleep(0.5)
+
+        while rotation_count < num_cal_rotations:
+            self._move_cw()
+            if self.testing:
+                # tell the fake home sensor that we have left home
+                self.home_sensor_pin.drive_low()
+                self._simulate_ticks(num_ticks=10)
+            # need to introduce a small pause before wait_for_active as we
+            # start from home position and want to make sure we clear the
+            # sensor before we start waiting for it to come around again
+            time.sleep(0.5)
```

Comment:
yea, this is the kind of thing I was hoping to work out at MQ obs. I was also thinking of having this be a parameter, because having long pauses is annoying when running the tests/fiddling with a breadboard setup.